stm/stm.ml

```diff
@@ -2214,7 +2214,10 @@ let new_doc { doc_type ; injections } =
   (* We must reset the whole state before creating a document! *)
   State.restore_root_state ();
 
-  let doc = VCS.init doc_type Stateid.initial (Vernacstate.Parser.init ()) in
+  let doc =
+    let ps = Vernacstate.Synterp.(freeze () |> parsing) in
```

Comment:
Why do this instead of `Pcoq.freeze()`?

Reply:
since we use Pcoq.unfreeze to unfreeze it would be more consistent to use Pcoq.freeze

Reply:
Indeed, that's a leftover of the first version of the patch that tried `Vernacstate.Synterp.freeze`

Reply:
Indeed it's been on my mind that maybe we should keep wrapping `Pcoq` somehow, I dunno?

Reply:
that doesn't sound useful IMO

Reply:
Replaced the calls to `Synterp.(freeze () > parsing`